Systematic build failure of ROOT master in the LCG DEBUG builds since Jan 15th

- [X] Checked for duplicates
None found

### Describe the bug
Since three nights (Jan 15th), the dev3 (ROOT-master) debug builds fail systematically with an error in
```
clang::ASTReader::ReadModuleMapFileBlock(llvm::SmallVector<unsigned long, 64u>&, clang::serialization::ModuleFile&, clang::serialization::ModuleFile const*, unsigned int)
```
See for example [this](http://cdash.cern.ch/upload/2db1c74842498c34342939e8d67598ae973967d8/ROOT-HEAD-build.log) .
<!--
A clear and concise description of what the wrong behavior is.
-->

### Expected behavior
A successful build ;-) .
<!--
A clear and concise description of what you expected to happen.
-->

### To Reproduce
LCG nightlies (lcgcmake). Did not try yet in standalone.
<!--
Steps to reproduce the behavior:
1. Your code that triggers the issue: at least a part; ideally something we can run ourselves.
2. Don't forget to attach the required input files!
3. How to run your code and / or build it, e.g. `root myMacro.C`, ...
-->

### Setup

1. ROOT master
2. Operating system: centos7, centos8
3. How you obtained ROOT, such as `dnf install` / binary download / you built it yourself: built from source (github)

### Additional context
None for the moment

The second cases you report seem unrelated at a first look:
```
In module 'Vc':
/build/jenkins/workspace/lcg_nightly_pipeline/install/dev3/Vc/1.4.2/x86_64-centos7-gcc11-opt/include/Vc/common/simdarray.h:501:18: error: 'Vc_1::SimdArray<type-parameter-0-0, N, type-parameter-0-2, N>::SimdArray<type-parameter-0-0, N, type-parameter-0-2, N>' from module 'Vc' is not present in definition of 'SimdArray<type-parameter-0-0, N, type-parameter-0-2, N>' in module 'Vc'
    Vc_INTRINSIC SimdArray(private_init_t, VectorType &&x) : data(std::move(x)) {}
                 ^
```

and

```
<<< cling interactive line includer >>>: fatal error: module file '/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/Vc.pcm' is out of date and needs to be rebuilt: signature mismatch
<<< cling interactive line includer >>>: note: imported by module 'MathCore' in '/cvmfs/sft-nightlies.cern.ch/lcg/nightlies/dev3/Tue/ROOT/HEAD/x86_64-centos7-gcc11-opt/lib/MathCore.pcm'
input_line_4:2:2: error: unknown type name 'include'
#include "TError.h"
 ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -504,11 +504,16 @@ install(
    PATTERN "pkgconfig" EXCLUDE
 )
 
+if(Vc_INCLUDE_DIR)
+  set(MODULES_ROOT_INCPATH "ROOT_INCLUDE_PATH=${Vc_INCLUDE_DIR}:${ROOT_INCLUDE_PATH}")
+endif()
+
 # modules.idx
 if(runtime_cxxmodules)
+
   add_custom_command(OUTPUT ${CMAKE_INSTALL_LIBDIR}/modules.idx
                      COMMAND ${CMAKE_COMMAND} -E remove -f modules.idx modules.timestamp
-                     COMMAND ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+                     COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
                              $<TARGET_FILE:root.exe> -l -q -b
                      WORKING_DIRECTORY ${CMAKE_INSTALL_LIBDIR}
@@ -530,7 +535,7 @@ if(WIN32)
                      DEPENDS $<TARGET_FILE:root.exe> Cling Hist Tree Gpad Graf HistPainter move_artifacts)
 else()
   add_custom_command(OUTPUT tutorials/hsimple.root
-                     COMMAND ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+                     COMMAND ${MODULES_ROOT_INCPATH} ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
                              ROOTIGNOREPREFIX=1 ROOT_HIST=0
                              $<TARGET_FILE:root.exe> -l -q -b -n -x hsimple.C -e return
                      WORKING_DIRECTORY tutorials

